src/wallet/wallet.h

```diff
@@ -1129,6 +1142,20 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
      * deadlock
      */
     void BlockUntilSyncedToCurrentChain();
+
+    /**
+     * Explicitly make the wallet learn the related scripts for outputs to the
+     * given key. This is purely for backward compatibility with earlier wallet
+     * versions, as CBasicKeyStore automatically does this implicitly for all
+     * keys now.
+     */
+    void LearnRelatedScripts(const CPubKey& key, OutputType = OUTPUT_TYPE_P2SH_SEGWIT);
```

Comment:
In commit "SegWit wallet support"

Maybe explain OUTPUT_TYPE_P2SH_SEGWIT default argument in the comment since it seems a little arbitrary. I guess if no OutputType parameter is passed you want to store the p2wpkh script unconditionally.

Reply:
Addressed by splitting the function up in two.